code/frameworks/nextjs/template/stories_default-js/Navigation.stories.jsx

```diff
@@ -0,0 +1,75 @@
+import { useRouter, usePathname, useSearchParams } from 'next/navigation';
+import React from 'react';
+
+function Component() {
+  const router = useRouter();
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+
+  const searchParamsList = Array.from(searchParams.entries());
+
+  const routerActions = [
+    {
+      cb: () => router.back(),
+      name: 'Go back',
+    },
+    {
+      cb: () => router.forward(),
+      name: 'Go forward',
+    },
+    {
+      cb: () => router.prefetch('/prefetched-html'),
+      name: 'Prefetch',
+    },
+    {
+      cb: () => router.push('/push-html', { forceOptimisticNavigation: true }),
+      name: 'Push HTML',
+    },
+    {
+      cb: () => router.refresh(),
+      name: 'Refresh',
+    },
+    {
+      cb: () => router.replace('/replaced-html', { forceOptimisticNavigation: true }),
+      name: 'Replace',
+    },
+  ];
+
+  return (
+    <div>
+      <div>pathname: {pathname}</div>
+      <div>
+        searchparams:{' '}
+        <ul>
+          {searchParamsList.map(([key, value]) => (
+            <li key={key}>
+              {key}: {value}
+            </li>
+          ))}
+        </ul>
+      </div>
+      {routerActions.map(({ cb, name }) => (
+        <div key={name} style={{ marginBottom: '1em' }}>
+          <button type="button" onClick={cb}>
+            {name}
+          </button>
+        </div>
+      ))}
+    </div>
+  );
+}
+
+export default {
+  component: Component,
+  parameters: {
+    nextAppDirectory: true,
+    nextNavigation: {
+      pathname: '/hello',
+      query: {
+        foo: 'bar',
+      },
+    },
+  },
+};
+
+export const Default = {};
```

Comment:
Could we use a play function to confirm the actions?

Reply:
Nice suggestion

Reply:
In this case, the action is not passed as an argument. Instead, the decorator uses `@storybook/action-addons` internally. It is not injected per dependency injection. So my question is, whether 

```
import { action } from '@storybook/addon-actions';
```

is mocked in general in all play functions.

@yannbf, maybe you know more.